### PR TITLE
Cache skeletons

### PR DESCRIPTION
**Pull request by goshakkk**
brunch/brunch#1054

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -116,26 +123,57 @@ var cleanURL = function(address) {
...
-    if (error != null) {
-      return callback(new Error("Temp dir error: " + error.toString()));
-    }
-    var cmd = 'git clone ' + url + ' ' + tempPath;
-    exec(cmd, function(error, stdout, stderr) {
-      if (error != null) {
-        return callback(new Error("Git clone error: " + stderr.toString()));
-      }
-      rimraf(sysPath.join(tempPath, '.git'), function(error) {
+  var cacheDir = sysPath.join(os.homedir(), '.init-skeleton', 'cache');
```
Comment (paulmillr) [inline index.js]: let's name this `.brunch`

File: index.js [hunk 2]
```diff
@@ -116,26 +123,57 @@ var cleanURL = function(address) {
...
+  var repoHash = sha1Digest(url);
```
Comment (paulmillr) [inline index.js]: we need to check commit hash too

Comment (goshakkk) [inline index.js]: Do we? I mean, we're literally cloning the whole repo and if the cached one exists, we do `git pull`

Comment (paulmillr) [inline index.js]: `git pull` is fine

**Pull request merged**